### PR TITLE
ci: verify all generated files are up-to-date

### PR DESCRIPTION
**Pull request by erikgb**
Closes https://github.com/cert-manager/trust-manager/issues/165

Also fixing leftovers (deepcopy gen) from https://github.com/cert-manager/trust-manager/pull/163 bymigrating from `deepcopy-gen` to `controller-gen` - which can also generate the needed code.

### CODE REVIEW

File: hack/verify-generated.sh
Comment (inteon) [inline hack/verify-generated.sh:23]: I remember that @wallrj had some issues with this git status setup.
I think at that point we settled for a script like the script here: https://github.com/cert-manager/approver-policy/blob/main/hack/util/verify.sh

And a verify target like this:
```make
verify:
	./hack/util/verify.sh $(MAKE) -s generate-all
```

Comment (inteon) [reply hack/verify-generated.sh:23]: I think that the issue was as follows:
- not considering untracked-files might mean that you miss files that have to be freshly created
- considering untracked-files means that local changes/ files will cause the check to fail, which was undesirable for development.

Not sure if I remember this 100% correctly.

File: hack/verify-generated.sh
Comment (SgtCoDFish) [inline hack/verify-generated.sh:24]: comment: As I'm sure you know, this is a stronger check than just checking if the generated files are out of date - but I think it's a check we should make anyway. Ideally files shouldn't be getting modified in CI so in CI it's good to verify they're not.

I might be wrong, but I think this would also fail if I the code generation was up-to-date, I'd edited something unrelated to code generation and I ran `make verify` locally, as part of development? That seems unfortunate - I use targets like `make verify` quite often when I'm developing a new feature.

This new way of validating would then mean I'd have to commit my changes before CI could pass. I suppose that's not the end of the world!

**Review by SgtCoDFish — APPROVED** [review]
/lgtm
/approve

Thanks! I think checking git has the downside of potentially interfering with local development, but it's better to have a solution for this than not!

**Comment by SgtCoDFish** [comment]
/hold

for tim's comments!

File: hack/verify-generated.sh
Comment (erikgb) [reply hack/verify-generated.sh:23]: I'll take a look and fix. Thanks for the pointers.

Comment (erikgb) [reply hack/verify-generated.sh]: I have now made a second attempt - based on the referenced work in approver-policy. PTAL!

File: hack/verify-generated.sh
Comment (erikgb) [reply hack/verify-generated.sh]: True, but outdated comment now. 😉

**Comment by erikgb** [comment]
@inteon There seem to have been something funky about the deepcopy gen, so I migrated to using controller-gen. Please let me know if I should put this is a separate PR.

**Comment by erikgb** [comment]
I have now fixed the excluded paths as noted in today's stand-up and updated the PR description. PTAL!

**Review by inteon — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**
